### PR TITLE
[New feature] P-CSCF address support

### PR DESCRIPTION
**Pull request by pruthvihebbani**
Summary:
Added P-CSCF IPv4 and IPv6 address support

Test Plan:
1.	Verified S1 SIM sanity
2.	Executed 2 new TCs – PR – #1577

### CODE REVIEW

File: lte/gateway/c/oai/tasks/sgw/pgw_pco.c [hunk 1]
```diff
@@ -315,20 +314,60 @@ int pgw_process_pco_link_mtu_request(
...
+int pgw_process_pco_pcscf_ipv6_address_req(
+    protocol_configuration_options_t* const pco_resp)
+{
+  if (!strlen((char*) spgw_config.pgw_config.pcscf.ipv6_addr.s6_addr)) {
+    OAILOG_ERROR(
+        LOG_SPGW_APP,
+        "PCO_CI_P_CSCF_IPV6_ADDRESS not configured. Ignoring the containerID "
+        "\n");
+    // Send P-CSCF IPv4 address if configured
+    if (RETURNok != pgw_process_pco_pcscf_ipv4_address_req(pco_resp)) {
```
Comment (uri200) [inline lte/gateway/c/oai/tasks/sgw/pgw_pco.c:355]: I am probably missing something, but how is the IPv4 sent?
Is it through pgw_pco_push_protocol_or_container_id?

File: lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py [hunk 2]
```diff
@@ -346,19 +347,27 @@ def sendDeactDedicatedBearerAccept(self, ue_id, bearerId):
...
-        # PDN Type = IPv4
-        req.pdnType_pr.pdn_type = 1
+        # PDN Type 1 = IPv4, 2 = IPv6, 3 = IPv4v6
+        req.pdnType_pr.pdn_type = pdn_type
         req.pdnAPN_pr.pres = 1
         req.pdnAPN_pr.len = len(apn)
         req.pdnAPN_pr.pdn_apn = (ctypes.c_ubyte * 100)(
             *[ctypes.c_ubyte(ord(c)) for c in apn[:100]]
         )
+        print("********* PDN type", pdn_type)
```
Comment (uri200) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py:365]: I made the same comment in the other diff, so probably i am wrong :D, but why don't you add an space or a dash?

`print("********* PDN type-", pdn_type)`

File: lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py [hunk 3]
```diff
@@ -346,19 +347,27 @@ def sendDeactDedicatedBearerAccept(self, ue_id, bearerId):
...
+        # Populate PCO if pcscf_addr_type is set
+        if pcscf_addr_type:
+            print("********* pcscf_addr_type", pcscf_addr_type)
```
Comment (uri200) [inline lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py:368]: Same here

File: lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py (hunk 2 above)
Comment (pruthvihebbani) [reply lte/gateway/python/integ_tests/s1aptests/s1ap_wrapper.py:365]: Space is automatically added. This is how the print looks - 
********* PDN type 1
********* pcscf_addr_type ipv4

File: lte/gateway/c/oai/tasks/sgw/pgw_pco.c (hunk 1 above)
Comment (pruthvihebbani) [reply lte/gateway/c/oai/tasks/sgw/pgw_pco.c:355]: If UE asks for P-CSCF IPv6 address and if it is not configured then we send IPv4 address. pgw_pco_push_protocol_or_container_id() populates the PCO structure with container id, length and contents and returns to the caller handle_s5_create_session_request() in pgw_handlers.c.  Later in sgw_handle_sgi_endpoint_created() PCO is copied to create_session_response_p message and sent to MME APP

**Review by uri200 — APPROVED** [review]
Looks good!

**Pull request closed without merging**